assets/js/agent/app.vue

```diff
@@ -56,27 +56,85 @@
     created: function() {
       this.loadUser();
       this.loadChannels();
+      this.syncSubscriptions();
     },
     methods: {
       loadChannels: function() {
         var $data = this.$data;
         var me = this;
 
         this.realmq.channels.list().then(function (channelList) {
-          channelList.items.sort(function (a, b) {
+          $data.channels = channelList.items.sort(function (a, b) {
             return new Date(a.createdAt) < new Date(b.createdAt) ? 1 : -1;
-          }).map(function(channel) {
-            channel.name = (channel.properties || {}).name || channel.id;
-            channel.name = channel.name.length > 30
-              ? channel.name.substr(0, 20) + '...'
-              : channel.name;
-            channel.dateFormatted = moment(channel.createdAt).format('LLL');
-          });
-          $data.channelList = channelList;
+          }).map(me.getChannelViewModel);
           me.activateChannel(channelList.count && channelList.items[0]);
         });
       },
 
+      getChannelViewModel: function(channel) {
+        var channelName = (channel.properties || {}).name || channel.id;
+        var viewModel = {
+          id: channel.id,
+          name: channelName.length > 30
+            ? channelName.substr(0, 20) + '...'
+            : channelName,
+          dateFormatted: moment(channel.createdAt).format('LLL'),
+          createdAt: channel.createdAt,
+          updatedAt: channel.updatedAt,
+        };
+
+        return viewModel;
+      },
+
+      onSubscriptionUpdated: function(subscription) {
+        var me = this;
+        var channels = this.channels;
+        var channelId = subscription.channelId;
+
+        this.realmq.channels.retrieve(channelId)
+          .then(function(channel) {
+            var existentChannel = channels.find(function(channel) {
+              return channel.id === channelId;
+            });
+
+            if (!existentChannel) {
+              existentChannel = me.getChannelViewModel(channel);
+              channels.unshift(existentChannel);
+            } else {
+              // todo update channel data.
```

Comment:
Something missing here?

Reply:
Not really.

Reply:
Either add a update of channel name or remove else branch completely.

Reply:
Yes: see 2f2e2f2